documentation/processing.tex

```diff
@@ -200,6 +202,10 @@ \subsubsection{Statement passing}
 
 These providers are activated when COPY instruction copies a file into the source code or when a macro is visited, respectively. They provide a sequence of statements to an arbitrary processor until all statements from the copy or macro definition are provided. After that, if there is no nested invocation, a provider with lower priority is selected.
 
+As HLASM allows macro recursion, the reader may think that macro definition provider may never finish so the main loop may never break. For this purpose, HLASM language itself has restriction for the level of nested macro invocation depending on the complexity of nested macros. We set the limit to 100 as it suffices in all tested code.
```

Comment:
"the reader may think" je podla mna take divne
```suggestion
To avoid infinite macro recursion, HLASM language itself has a restriction for the level of nested macro invocation depending on the complexity of nested macros. We set the limit to 100 as it suffices in all tested code.
```